### DISCUSSION TITLE
Plain text request body

### DISCUSSION THREAD

**Discussion by thobson** [swaggest/rest#150] [Q&A] [answered]
Firstly, thanks for your work on this.

I need to accept text/plain and text/html request bodies i.e. not JSON/XML. I can achieve this using the `SetRequest` function:

```go
type DocumentInput struct {
    Tenancy string `path:"tenancy"`
    Content string 
}

func (input *DocumentInput) SetRequest(r *http.Request) {
    b, _ := ioutil.ReadAll(r.Body)
    r.Body.Close()
    input.Content = string(b)
}
```

However the generated swagger docs are not aware of the plain text payload:

```json
{
    "/{tenancy}/documents/": {
        "post": {
            "parameters": [
                {
                    "name": "tenancy",
                    "in": "path",
                    "required": true,
                    "schema": {
                        "type": "string"
                    }
                }
            ]
        }
    }
}
```

Is there any way I can tell the Interactor that the request body should be a text payload?

Thanks

**Comment by vearutop** [comment, marked as answer]
Hello, in such case, you can use handler annotation when adding your use case to router/web service.

Please check an [example](https://github.com/swaggest/rest/blob/v0.2.44/_examples/advanced-generic/router.go#L193).

```go
s.Post("/{tenancy}/documents/", textReqBody(), nethttp.RequestBodyContent("text/plain"))
```
